Makefile.common_app

```diff
@@ -0,0 +1,103 @@
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#*******************************************************************************
+
+COMMON_APP_FLAGS = 0x000
+
+#####################################################################
+#                            BLUETOOTH                              #
+#####################################################################
+ifeq ($(ENABLE_BLUETOOTH), 1)
+    COMMON_APP_FLAGS = 0x200  # APPLICATION_FLAG_BOLOS_SETTINGS
+    COMMON_APP_DEFINES += HAVE_BLE BLE_COMMAND_TIMEOUT_MS=2000 HAVE_BLE_APDU
+    COMMON_SDK_SOURCE_PATH += lib_blewbxx lib_blewbxx_impl
+endif
+
+#####################################################################
+#                               DEBUG                               #
+#####################################################################
+ifneq ($(DEBUG), 0)
+    COMMON_APP_DEFINES += HAVE_PRINTF
+    ifeq ($(TARGET_NAME),TARGET_NANOS)
+        COMMON_APP_DEFINES += PRINTF=screen_printf
+    else
+        COMMON_APP_DEFINES += PRINTF=mcu_usb_printf
+    endif
+else
+        COMMON_APP_DEFINES += PRINTF\(...\)=
+endif
+
+#####################################################################
+#                        IO SEPROXY BUFFER SIZE                     #
+#####################################################################
+ifneq ($(DISABLE_DEFAULT_IO_SEPROXY_BUFFER_SIZE), 1)
+    ifeq ($(TARGET_NAME), TARGET_NANOS)
+        COMMON_APP_DEFINES += IO_SEPROXYHAL_BUFFER_SIZE_B=128
+    else
+        COMMON_APP_DEFINES += IO_SEPROXYHAL_BUFFER_SIZE_B=300
+    endif
+endif
+
+#####################################################################
+#                          COMMON DEFINES                           #
+#####################################################################
+COMMON_APP_DEFINES += $(DEFINES_LIB)
+COMMON_APP_DEFINES += APPNAME=\"$(APPNAME)\"
+COMMON_APP_DEFINES += APPVERSION=\"$(APPVERSION)\"
+COMMON_APP_DEFINES += MAJOR_VERSION=$(APPVERSION_M) MINOR_VERSION=$(APPVERSION_N) PATCH_VERSION=$(APPVERSION_P)
+COMMON_APP_DEFINES += OS_IO_SEPROXYHAL
+
+ifneq ($(DISABLE_COMMON_APP_DEFINES), 1)
+    COMMON_APP_DEFINES += HAVE_SPRINTF HAVE_SNPRINTF_FORMAT_U
+    COMMON_APP_DEFINES += HAVE_IO_USB HAVE_L4_USBLIB IO_USB_MAX_ENDPOINTS=6 IO_HID_EP_LENGTH=64 HAVE_USB_APDU
+    COMMON_APP_DEFINES += USB_SEGMENT_SIZE=64
+    COMMON_APP_DEFINES += BLE_SEGMENT_SIZE=32
+    COMMON_APP_DEFINES += HAVE_WEBUSB WEBUSB_URL_SIZE_B=0 WEBUSB_URL=""
+    COMMON_APP_DEFINES += UNUSED\(x\)=\(void\)x
+    
+    COMMON_SDK_SOURCE_PATH += lib_stusb lib_stusb_impl
```

Comment:
What do you think of adding this by default in SDK `Makefile.rules` with a disabler.
Same for lib_ux below.

Somethink like:
```
ifneq ($(DISABLE_DEFAULT_SDK_LIB), 1)
SDK_SOURCE_PATH += lib_stusb lib_stusb_impl
ifeq ($(TARGET_NAME),TARGET_STAX)
SDK_SOURCE_PATH += lib_nbgl lib_ux_stax qrcode
else
SDK_SOURCE_PATH += lib_bagl lib_ux
endif
endif
```

So that it default to True (99% of cases) and is located all in the same dir.
We just need to check that it will only break very few apps (does with custom lib_stusb).
What do you think about it?

Reply:
I like the idea, I'll need to find a way measure the impact :)

Reply:
its breaking 22 apps :(

Reply:
Can you share a small list, that seems a lot! :open_mouth:

Reply:
https://github.com/LedgerHQ/ledger-secure-sdk/actions/runs/4627192147